utilities/shared_disk.py

```
@@ -0,0 +1,144 @@
+        vm2_name (str): Name of the second VM (consumer).
+        vm_ssh_connections (SSHConnectionManager): SSH connection manager.
+        source_provider_data (dict[str, Any]): Provider configuration from .providers.json.
+        vm1_info (dict[str, Any]): VM1 source data including OS type.
+        vm2_info (dict[str, Any]): VM2 source data including OS type.
+        shared_disk_device (str): Shared disk device path. Defaults to "/dev/vdc".
+
+    Raises:
+        AssertionError: If shared disk data verification fails.
+        RuntimeError: If SSH commands fail.
+    """
+    LOGGER.info(f"Verifying shared disk between {vm1_name} and {vm2_name}")
```

Ruff (0.15.9)

[warning] Logging statement uses f-string

(G004)